pkg/prometheus/promcfg.go

```diff
@@ -3001,12 +3001,12 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 	if len(sc.Spec.EC2SDConfigs) > 0 {
 		configs := make([][]yaml.MapItem, len(sc.Spec.EC2SDConfigs))
 		for i, config := range sc.Spec.EC2SDConfigs {
-			if config.Region != nil {
-				configs[i] = append(configs[i], yaml.MapItem{
-					Key:   "region",
-					Value: config.Region,
-				})
-			}
+			configs[i] = cg.addProxyConfigtoYaml(configs[i], s, config.ProxyConfig)
+
+			configs[i] = append(configs[i], yaml.MapItem{
```

Comment:
why remove the `config.Region != nil` test?

Reply:
Ah, I had a bit of back and forth with making the field required or optional. Let me fix it.